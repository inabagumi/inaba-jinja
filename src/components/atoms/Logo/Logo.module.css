.logo {
  display: inline-block;
  fill: currentColor;
  height: 1em;
  line-height: 1;
  vertical-align: middle;
  width: auto;
}

.logoVertical {
  height: auto;
  width: 1em;
}
