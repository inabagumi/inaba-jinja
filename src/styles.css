:root {
  --ij-default-font-family: -apple-system, BlinkMacSystemFont,
    Helvetica Neue, Helvetica, Arial, YuGothic, Yu Gothic, sans-serif;
  --ij-serif-font-family: Garamond, Times New Roman, YuMincho, Yu Mincho,
    serif;
}

html {
  box-sizing: border-box;
  font-family: var(--ij-default-font-family);
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  -webkit-text-size-adjust: none;
}

body {
  background-color: #424242;
  color: #fafafa;
  margin: 0;
  line-height: 2;
}

*,
*::before,
*::after {
  box-sizing: inherit;
}

body,
html,
#__next {
  height: 100%;
}

p {
  margin: 0;
  padding: 0;
}

a {
  color: #f57f17;
}

a:hover {
  color: #ff6f00;
}
