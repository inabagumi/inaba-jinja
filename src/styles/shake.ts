import { keyframes } from '@emotion/react'

const shake = keyframes`
  from {
    transform: translateY(0) rotate(180deg);
  }

  to {
    transform: translateY(20px) rotate(170deg);
  }
`

export default shake
