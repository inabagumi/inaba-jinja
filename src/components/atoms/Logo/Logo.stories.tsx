import { storiesOf } from '@storybook/react'
import React from 'react'
import Logo from './Logo'

storiesOf('Logo', module).add('default', () => <Logo height="70" width="250" />)
