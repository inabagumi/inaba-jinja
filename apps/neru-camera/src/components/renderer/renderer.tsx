import { makeStyles } from '@material-ui/core/styles'
import { Application, Sprite } from 'pixi.js'
import { Viewport } from 'pixi-viewport'
import React, {
  ReactElement,
  SyntheticEvent,
  forwardRef,
  useCallback,
  useEffect,
  useImperativeHandle,
  useRef,
  useState
} from 'react'
import { Asset } from '../../context/asset-context'
import { ChromaKeyFilter } from '../../filters/ChromaKeyFilter'

const useStyles = makeStyles({
  container: {
    position: 'relative',
    width: '100%'
  },
  preloaded: {
    display: 'none'
  },
  preview: {
    display: 'block',
    height: 'auto',
    width: '100%'
  }
})

export type RefObject = {
  toBlob: (type: string, quality: number) => Promise<Blob>
}

export type Props = {
  asset: Asset
  cameraStream: MediaStream
}

const Renderer = forwardRef<RefObject, Props>(
  ({ asset, cameraStream }, ref): ReactElement => {
    const [viewport, setViewport] = useState<Viewport>()
    const classes = useStyles()

    const previewRef = useRef<HTMLCanvasElement>(null)
    const cameraRef = useRef<HTMLVideoElement>(null)
    const overlayRef = useRef<HTMLVideoElement>(null)

    const setup = useCallback(
      (event: SyntheticEvent<HTMLVideoElement, Event>) => {
        if (!previewRef.current) return

        const background = Sprite.from(event.currentTarget)

        const app = new Application({
          height: background.height,
          preserveDrawingBuffer: true,
          view: previewRef.current,
          width: background.width
        })
        const viewport = new Viewport({
          interaction: app.renderer.plugins.interaction,
          screenHeight: app.screen.height,
          screenWidth: app.screen.width,
          worldHeight: app.screen.height,
          worldWidth: app.screen.width
        })

        viewport
          .drag()
          .pinch()
          .wheel()
          .decelerate()

        app.stage.addChild(background)
        app.stage.addChild(viewport)

        setViewport(viewport)

        if (overlayRef.current) overlayRef.current.load()
      },
      []
    )

    const handleLoadedData = useCallback(
      (event: SyntheticEvent<HTMLVideoElement, Event>) => {
        if (!viewport) return

        const target = event.currentTarget
        const overlay = Sprite.from(target)
        const { keyColor } = target.dataset

        overlay.filters = [new ChromaKeyFilter(keyColor)]
        overlay.anchor.set(0.5)
        overlay.position.set(
          viewport.worldWidth * 0.5,
          viewport.worldHeight * 0.5
        )

        viewport.addChild(overlay)
      },
      [viewport]
    )

    useEffect(() => {
      if (!cameraRef.current) return

      cameraRef.current.srcObject = cameraStream
    }, [cameraStream])

    useImperativeHandle(ref, () => ({
      toBlob: (type = 'image/png', quality = 1.0): Promise<Blob> =>
        new Promise((resolve, reject): void => {
          if (!previewRef.current) {
            reject(new TypeError('Canvas is not mounted.'))
            return
          }

          const canvas = document.createElement('canvas')
          const context = canvas.getContext('2d')

          canvas.width = previewRef.current.width
          canvas.height = previewRef.current.height

          if (!context) {
            reject(new TypeError('Context does not exist.'))
            return
          }

          try {
            context.drawImage(previewRef.current, 0, 0)
            canvas.toBlob(
              blob => {
                if (!blob) {
                  reject(new TypeError('Blob is empty.'))
                  return
                }

                resolve(blob)
              },
              type,
              quality
            )
          } catch (error) {
            reject(error)
          }
        })
    }))

    return (
      <div className={classes.container}>
        <canvas className={classes.preview} ref={previewRef} />

        <video
          className={classes.preloaded}
          crossOrigin="anonymouse"
          muted
          onLoadedData={setup}
          playsInline
          ref={cameraRef}
        />
        <video
          className={classes.preloaded}
          crossOrigin="anonymouse"
          data-key-color={asset.keyColor}
          loop
          muted
          onLoadedData={handleLoadedData}
          playsInline
          preload="none"
          ref={overlayRef}
          src={asset.src}
        />
      </div>
    )
  }
)

export default Renderer
