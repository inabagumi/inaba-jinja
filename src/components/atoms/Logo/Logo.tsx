import clsx from 'clsx'
import React, { FC, SVGProps } from 'react'
import HorizontalLogo from '../../../assets/logo.svg'
import VerticalLogo from '../../../assets/vertical-logo.svg'
import styles from './Logo.module.css'

type Props = {
  vertical?: boolean
} & SVGProps<SVGSVGElement>

const Logo: FC<Props> = ({ vertical = false, ...props }) => {
  const RawLogo = vertical ? VerticalLogo : HorizontalLogo
  const className = clsx(
    styles.logo,
    {
      [styles.logoVertical]: vertical
    },
    props.className
  )

  return <RawLogo {...props} className={className} />
}

export default Logo
