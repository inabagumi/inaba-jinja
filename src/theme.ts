import deepOrange from '@material-ui/core/colors/deepOrange'
import orange from '@material-ui/core/colors/orange'
import { createMuiTheme } from '@material-ui/core/styles'

export default createMuiTheme({
  palette: {
    primary: deepOrange,
    secondary: orange,
    type: 'light'
  },
  typography: {
    body1: {
      lineHeight: 2
    },
    body2: {
      lineHeight: 2
    },
    fontFamily: [
      'Roboto',
      'Helvetica',
      'Arial',
      'Noto Sans JP',
      'sans-serif'
    ].join(',')
  }
})
