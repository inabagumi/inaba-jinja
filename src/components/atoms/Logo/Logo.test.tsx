/* eslint-env jest */

import { ThemeProvider } from '@material-ui/core/styles'
import React, { FC } from 'react'
import { IntlProvider } from 'react-intl'
import renderer from 'react-test-renderer'
import theme from '../../../theme'
import Logo from './Logo'

const App: FC = ({ children }) => (
  <IntlProvider locale="en">
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  </IntlProvider>
)

describe('Logo', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <App>
          <Logo />
        </App>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
