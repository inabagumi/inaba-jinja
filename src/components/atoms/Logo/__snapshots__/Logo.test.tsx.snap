// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders correctly 1`] = `
<SvgMock
  className="jsx-1961443691"
  focusable="false"
  role="img"
/>
`;
